assignment1_calculator.c

```diff
@@ -0,0 +1,125 @@
+// Kalpvriksha Program | Batch-2025: C Programming Assignments covering Calculator.
+
+// assignment 1: Calculator, input format :- 3+5
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+
+#define MAX_EXPR_LEN 100
+
+typedef enum {
+    NO_ERROR,
+    DIVISION_BY_ZERO,
+    INVALID_EXPRESSION
+} ErrorType;
+
+ErrorType error = NO_ERROR;
+
+char* skip_whitespace(char *expr) {
+    while (*expr == ' ') expr++;
+    return expr;
+}
+
+int evaluate_term(char **expr);
+
+int evaluate_expression(char **expr);
+
+int parse_integer(char **expr) {
+    *expr = skip_whitespace(*expr);
+    int result = 0;
+    while (isdigit(**expr)) {
```

Comment:
Avoid using inbuilt functions unless it is system calls, please change the built-in functions to user implementation function. take care at all places.